lib/riffed/client.ex

```diff
@@ -209,6 +248,8 @@ defmodule Riffed.Client do
             call_thrift(new_client, call_name, args, retry_count + 1)
           err = {:error, _} ->
             {new_client, err}
+          exception = {:exception, exception_record} ->
```

Comment:
I've found that this isn't enough to catch the exception.. The actual erlang thrift client will `throw` the exception, not just return it. So when working with services that go through the thrift client (maybe the tests here don't?), the `GenServer` will `EXIT` before we reach this `case` statement

https://github.com/apache/thrift/blob/master/lib/erl/src/thrift_client.erl#L150
```erlang
        [Exception] ->
            throw({NewClient, {exception, Exception}})
```

I had to modify the `:thrift_client.call` to look something like this:
```elixir
        {thrift_client, response} = try do
          :thrift_client.call(client.client, call_name, args)
        catch
          exception_response = {thrift_client, {:exception, exception}} -> exception_response
        end
```